packages/fast-components/src/slider/slider.ts

```diff
@@ -0,0 +1,327 @@
+            this.updateForm();
+        }
+
+        const percentage: number =
+            this.direction !== "rtl"
+                ? (1 - (Number(this.value) / this.max - this.min)) * 100
+                : (Number(this.value) / this.max - this.min) * 100;
+
+        this.position = this.isDragging
+            ? `right: ${percentage}%; transition: all 0.1s ease;`
+            : `right: ${percentage}%; transition: all 0.2s ease;`;
+
+        this.$emit("change");
+    }
+
+    /**
+     * Min allowed value
+     */
+    @attr
+    public min: number = 0; // Map to proxy element.
+    private minChanged(): void {
+        if (this.proxy instanceof HTMLElement) {
+            this.proxy.min = `${this.min}`;
+        }
+    }
+
+    /**
+     * Max allowed value
+     */
+    @attr
+    public max: number = 10; // Map to proxy element.
+    private maxChanged(): void {
+        if (this.proxy instanceof HTMLElement) {
+            this.proxy.max = `${this.max}`;
+        }
+    }
+
+    @attr
+    public step: number = 1; // Map to proxy element.
+    private stepChanged(): void {
+        if (this.proxy instanceof HTMLElement) {
+            this.proxy.step = `${this.step}`;
+        }
+    }
+
+    /**
+     * Orientation value, horizontal or vertical
+     */
+    @attr
+    public orientation: SliderOrientation = SliderOrientation.horizontal;
+
+    /**
+     * mode value, singleValue | dualThumb | adjustFromUpper | adjustFromLower
+     */
+    @attr
+    public mode: SliderMode = SliderMode.singleValue;
+
+    /**
+     * Set to true when the component has constructed
+     */
+    private constructed: boolean = false;
+    protected proxy = document.createElement("input");
+
+    constructor() {
+        super();
+
+        this.proxy.setAttribute("type", "range");
+        this.setAttribute("role", "slider");
+        this.setAttribute("tabindex", "0");
+        this.direction = this.getDirection();
+        this.constructed = true;
+    }
+
+    public connectedCallback(): void {
+        super.connectedCallback();
+        this.updateForm();
+        this.setupTrackConstraints();
+        this.setupListeners();
+        this.setupClassForOrientation();
+        this.setupDefaultValue();
+    }
+
+    public disconnectedCallback(): void {
+        this.removeEventListener("keydown", this.keypressHandler);
+        this.removeEventListener("mousedown", this.clickHandler);
+    }
+
+    protected keypressHandler = (e: KeyboardEvent) => {
+        super.keypressHandler(e);
+        switch (e.keyCode) {
+            case keyCodeArrowRight:
+            case keyCodeArrowUp:
+                this.increment();
+                break;
+            case keyCodeArrowLeft:
+            case keyCodeArrowDown:
+                this.decrement();
+                break;
+        }
+    };
+
+    private getDirection = (): string => {
+        const dirNode: any = this.parentElement!.closest("div[dir]");
```

Comment:
A couple of questions here:

1. Is this `any`? I think we'd have a typing for this.
2. You're looking for a div, that seems like a strong assumption. I think you likely want to traverse the DOM to check for a dir attribute.

Reply:
Ultimately, what happens if I set the dir attribute on the `body` and leave it. I think this would fail.

Reply:
This is how it worked before, I'm not sure it would work any less here

Reply:
ok, so it turns out I didn't need to put div[dir], I think just [dir] will traverse up as far as you can traverse manually.

Reply:
Also if that is the case I think that justifies using any, unless I can use HTMLElement or something

Reply:
Looks like `Element | null` may be the best typing: https://developer.mozilla.org/en-US/docs/Web/API/Element/closest.